### PR TITLE
Move all lua files to src/ folder

### PR DESCRIPTION
**Pull request by Zeimyth**
So, it's always bothered me how unorganized and flat the source code is for this repo. I have a few things I am thinking about trying to improve, but I felt like I needed to start with this cosmetic change.

The files could be cleaned up more than just all going into the src/ directory, of course, but I felt like this was probably the most significant change and moving things into sub-folders from here will be easier. I don't know what might break by moving these files (though the server and client both run and pass some basic manual testing), so I wanted to get this opened as soon as possible to both a) get feedback about the changes and b) avoid merge conflicts.

The commands to run the client (`love ./`) and server (`luajit server.lua`) still work without needing modification. I don't know what building/deploying commands are relied on, so I wasn't able to verify those.

### CODE REVIEW

File: conf.lua [hunk 1]
```diff
@@ -1,6 +1,10 @@
-require("consts")
+require("src/consts")
+-- require("src/logger")
 
 function love.conf(t)
+  -- logger.error("Some constants:")
```
Comment (Zeimyth) [inline conf.lua:5]: oh oops, I didn't realize these made it through. I'll clean them up.

File: src/main.lua [hunk 2]
```diff
@@ -0,0 +1,174 @@
+require("src/developer")
```
Comment (Zeimyth) [inline src/main.lua:1]: This file was moved just like all of the other ones, but it has so many import changes that git must be detecting it as a new file. Weird.

File: src/scores.lua [hunk 3]
```diff
@@ -1,3 +1,5 @@
+require("src/globals")
```
Comment (Zeimyth) [inline src/scores.lua:1]: After moving, this file started complaining. I'm not sure what about the move caused things to change, but transitive requires are probably an antipattern anyways, so...

File: conf.lua (hunk 1 above)
Comment (Zeimyth) [reply conf.lua]: These have been removed.

**Comment by JamesVanBoxtel** [comment]
I'm torn about this... it would be nice to have things organized better, but its also a huge pain to move everything and deal with merge fallout and making sure all the themes / image / build scripts / server work correctly

**Comment by Zeimyth** [comment]
One thing I can say is that all of the vanilla assets are working without any issues. It seems like everything is referenced based on the root directory, so the Lua scripts didn't need to be modified at all to keep reading them. As far as building, we already had some Lua files in their own folders (the engine subdirectory) and that was working before, so it's possible that everything will work still.

In terms of merge conflicts, main is the most likely to have problems. The other files should move seamlessly. The only problem might be requires or new files that are added in other branches.

What build commands are there? I am happy to test what I can to help increase confidence. Besides that, I can rebase when other changes are merged to beta to try to stay on top of any conflicts while we decide what we're comfortable with.

**Comment by Endaris** [comment]
I don't think there's much value in moving everything over because the project isn't as big and we implement folder structures as we implement new things or refactor old ones.
Most notably we have an overhaul of all the UI stuff coming and it would make a lot more sense to do this step by step during that process.
Conversely I also feel like it's much easier to keep the source files on the root level unless there are sensible groupings (such as engine, graphics, computerPlayer, server etc) and instead put the other assets into extra folders. As those hardly receive any updates, moving them to folders like readme, assets is much less likely to provoke merge conflicts and if there are any, those will be much easier to resolve.

I'd be much more interested in the things you want to improve and why this change would be necessary for that because as it is, this already affects various bigger projects in the making, most notably #502 and #328 that aren't going to be merged soon but will be strongly affected by this. So from my point of view, this change mainly creates some hours of work, especially for Sankyr, without tangible benefits

**Pull request closed without merging**